prog/probe.go

```diff
@@ -213,40 +213,42 @@ func probeMain(flags probeFlags, targets []appclient.Target) {
 	}
 
 	p := probe.New(flags.spyInterval, flags.publishInterval, clients, flags.noControls)
+	var processCache *process.CachingWalker
 
-	hostReporter := host.NewReporter(hostID, hostName, probeID, version, clients, handlerRegistry)
-	defer hostReporter.Stop()
-	p.AddReporter(hostReporter)
-	p.AddTagger(probe.NewTopologyTagger(), host.NewTagger(hostID))
+	if flags.kubernetesRole != kubernetesRoleCluster {
+		hostReporter := host.NewReporter(hostID, hostName, probeID, version, clients, handlerRegistry)
+		defer hostReporter.Stop()
+		p.AddReporter(hostReporter)
+		p.AddTagger(probe.NewTopologyTagger(), host.NewTagger(hostID))
 
-	var processCache *process.CachingWalker
-	if flags.procEnabled {
-		processCache = process.NewCachingWalker(process.NewWalker(flags.procRoot, false))
-		p.AddTicker(processCache)
-		p.AddReporter(process.NewReporter(processCache, hostID, process.GetDeltaTotalJiffies, flags.noCommandLineArguments))
-	}
+		if flags.procEnabled {
+			processCache = process.NewCachingWalker(process.NewWalker(flags.procRoot, false))
+			p.AddTicker(processCache)
+			p.AddReporter(process.NewReporter(processCache, hostID, process.GetDeltaTotalJiffies, flags.noCommandLineArguments))
+		}
 
-	dnsSnooper, err := endpoint.NewDNSSnooper()
-	if err != nil {
-		log.Errorf("Failed to start DNS snooper: nodes for external services will be less accurate: %s", err)
-	} else {
-		defer dnsSnooper.Stop()
-	}
+		dnsSnooper, err := endpoint.NewDNSSnooper()
+		if err != nil {
+			log.Errorf("Failed to start DNS snooper: nodes for external services will be less accurate: %s", err)
+		} else {
+			defer dnsSnooper.Stop()
+		}
 
-	endpointReporter := endpoint.NewReporter(endpoint.ReporterConfig{
-		HostID:       hostID,
-		HostName:     hostName,
-		SpyProcs:     flags.spyProcs,
-		UseConntrack: flags.useConntrack,
-		WalkProc:     flags.procEnabled,
-		UseEbpfConn:  flags.useEbpfConn,
-		ProcRoot:     flags.procRoot,
-		BufferSize:   flags.conntrackBufferSize,
-		ProcessCache: processCache,
-		DNSSnooper:   dnsSnooper,
-	})
-	defer endpointReporter.Stop()
-	p.AddReporter(endpointReporter)
+		endpointReporter := endpoint.NewReporter(endpoint.ReporterConfig{
+			HostID:       hostID,
+			HostName:     hostName,
+			SpyProcs:     flags.spyProcs,
+			UseConntrack: flags.useConntrack,
+			WalkProc:     flags.procEnabled,
+			UseEbpfConn:  flags.useEbpfConn,
```

Comment:
Since this now only executes in the non-k8s-cluster-role case, you can get rid of the code that clears all these flags in https://github.com/weaveworks/scope/blob/71972c302ad9612b243a47fadfeb9489800e7157/prog/probe.go#L115.

Reply:
Yeah, I thought it was better left as-is than to cherry-pick those that are used now. It’s clearer.

Reply:
but _none_ of the flags set earlier on are used anymore, except for the two that are clearly kubernetes related, i.e.
```
		flags.kubernetesKubeletPort = 0
		flags.kubernetesEnabled = true
```

Reply:
I guess the fact that we are having this debate proves my point, but still, I see `spyProcs` cleared on line 118 and used on line 126.